### DISCUSSION TITLE
Libvips error

### DISCUSSION THREAD

**Discussion by Jimcoddington10** [libvips/libvips#4315] [Q&A]
I am making a mosaic of x-radiographs. The mosaic balance function works on some of the assemblies and not on others. I cannot see any difference in the images and they were all acquired the same way. Some overlaps are very fine but the assembly looks good even in those cases. They are all one point mosaics. Here is the error message ( I think...)
![Untitled](https://github.com/user-attachments/assets/1d44dc33-f73e-4bbd-9d23-f741e7b703fb)

**Comment by jcupitt** [comment]
Hi @Jimcoddington10, nice to hear from you!

The error sounds strange. Is there any way you could get the files to me so I can try to reproduce this? Maybe upload a zip with a password to dropbox, or even mail me a thumbdrive?

**Comment by Jimcoddington10** [comment]
John, Thanks, good to hear from you as well. It turns out I am really rusty on VIPS....which is to say that I neglected to save my workspace, which required some getting these images (I did not take them) properly oriented to one another before mosaicing. Hence I don't have images to send you that you would be able to make sense of for a mosaic,  nor do I have a WS that you could use as guidance for that. So I will do it again, see if the error message comes up again and if so send them along to you for further investigating.But the above does raise a question of why there is not a back up of the ws file. The proper box is checked in preferences but maybe there is something more direct one needs to do to activate an auto back up?Hope you all are well John and Happy Holidays.

**Reply by jcupitt** [reply]
There should be a backup, I wonder what went wrong.

You can't fix images and then mosaic them, for complicated technical reasons. You need to load your input images, do any cropping / flipping / rotating, then save them all again.

Once that's done, quit, start nip2 again, load the corrected images, and mosaic them.

nip4 is almost done, did you see?

https://github.com/jcupitt/nip4

It's supposed to be quicker and easier to use (and more modern).

There's a little video of the current state on the libvips news page:

https://www.libvips.org/2024/11/26/nip4-November-progress.html

**Comment by Jimcoddington10** [comment]
John,I sent three of the radiographs to you via dropbox at your gmail account. Hope that is still active but if not please advise as to where to send them. Those three, when mosaiced, give the error message. Attached is the WS as well as I needed to flip the images to get them properly oriented to the painting in reality so I am hoping you can edit it as necessary to the path where you put the images. Let me know if I have placed any insurmountable hurdles in the way.
Best,

Jim

**Reply by jcupitt** [reply]
I put a `final.tif` in that folder, together with `prepare-frames.ws` and `join.ws`.

`prepare-frames.ws` spins the three frames upright and crops any white bits from the border. Save the three images as `left.v`, `centre.v` and `right.v`.

`join.ws` loads left, centre and right, joins them up and rebalances them. 

`final.tif` is the result image, saved as a compressed 16-bit TIFF. Hopefully it'll load anywhere, but let me know if it doesn't.

**Reply by jcupitt** [reply]
... I used nip4 to make the mosaic, hopefully the workspace files will load into nip2, though I've not tested it.

**Comment by Jimcoddington10** [comment]
Thanks John. And no error? Odd. I think I will go to nip4 :)

Jim
________________________________
